drivers/sensor/lsm6dsl/lsm6dsl_trigger.c

```diff
@@ -49,6 +49,11 @@ int lsm6dsl_trigger_set(const struct device *dev,
 
 	__ASSERT_NO_MSG(trig->type == SENSOR_TRIG_DATA_READY);
 
+	if (!drv_data->irq_dt_defined) {
```

Comment:
Why not use `drv_data->gpio == NULL` as the conditional here instead of a dedicated flag?

Reply:
I thought that a flag with a specific name was more clear. OK, I'll do in this way and I'll a comment. Thanks!